src/NUnitFramework/framework/Constraints/ThrowsConstraint.cs

```diff
@@ -161,10 +161,9 @@ internal static Exception Intercept(object invocation)
                 {
                     using (var region = AsyncInvocationRegion.Create(invocationDescriptor.Delegate))
                     {
-                        object result = invocationDescriptor.Invoke();
-
                         try
                         {
+                            object result = invocationDescriptor.Invoke();
```

Comment:
Moved the `Invoke()` inside the try, because the operation can fail synchronously (i.e. throw before the first await, or before returning a task)